src/misc_gui.cpp

```diff
@@ -780,6 +780,25 @@ static int GetCaretWidth()
 	return GetCharacterWidth(FS_NORMAL, '_');
 }
 
+/**
+ * Reposition edit text box rect based on textbuf length can caret position.
+ * @param r Initial rect of edit text box.
+ * @param tb The Textbuf being processed.
+ * @return Updated rect.
+ */
+static Rect ScrollEditBoxTextRect(Rect r, const Textbuf &tb)
+{
+	const int linewidth = tb.pixels + GetCaretWidth();
+	const int boxwidth = r.Width();
+	if (linewidth <= boxwidth) return r;
+
+	/* Extend to cover whole string. This is left-aligned, adjusted by caret position. */
+	r = r.WithWidth(linewidth, false);
+
+	/* Slide so that the caret is at the centre, limited by . */
```

Comment:
Limited by what? Literally `.`, if so maybe use `'.'` to make clear it's not intended to be read as normal punctuation.

Reply:
Limited by me not finishing the sentence at all.